mk/rules.mk

```diff
@@ -20,6 +20,24 @@ $(BUILD_DIR)/%.a: $(CONFIG)
 	$(Q)rm -f $@
 	$(Q)$(CC_AR) rcs $@ $(filter %.o,$^)
 
+# NOTE: Checking if there's duplicated symbols by actually linking
```

Comment:
This piece of Makefile is a bit crypto -- can you expand on what is happening here?

Reply:
So after discuss with @mkannwischer, I think we want to make sure that we properly namespace everything, which means there's no duplicated symbols in the built library.

However as I mentioned in the PR description, we cannot check this with `$AR`, we need to actually use it to find out if there's duplicated symbol.

Therefore the easiest way I can think of is to create a dummy c file on the fly, pipe it into `$CC` and link with the built library for checking the duplicated symbol.

Btw, We can have a separate make target for doing this. I added this here is because I thought that it'd be nice if the build system can complain about duplicated symbols immediately.